core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -473,7 +480,7 @@ protected Location onReplicaShard(BulkShardRequest request, IndexShard indexShar
                 throw new IllegalStateException("Unexpected index operation: " + item.request());
             }
         }
-        return location;
+        return new WriteResult<>(null, location);
```

Comment:
changed, now we have `ReplicaWriteResult` for handling the replica case, so no need for passing null for the response.